File: julia/skmoefs/fuzzysets/UniverseFuzzySet.jl
"""
    Universe Fuzzy Set
"""

import Base.show

mutable struct UniverseFuzzySet
end

function __init__(self::UniverseFuzzySet)
end

show(io::IO, self::UniverseFuzzySet) = print(io,
    "Universe Fuzzy Set"
)

function isInSupport(self::UniverseFuzzySet, x)
    return true
end

function isFirstOfPartition(self::UniverseFuzzySet)
    return true
end

function isLastOfPartition(self::UniverseFuzzySet)
    return true
end

function membershipDegree(self::UniverseFuzzySet,x)
    return 1.0
end

function createFuzzySet()
    return UniverseFuzzySet()
end

File: python/skmoefs/fuzzysets/UniverseFuzzySet.py
from __future__ import division
from skmoefs.fuzzysets.FuzzySet import FuzzySet

class UniverseFuzzySet(FuzzySet):
    def __init__(self):
        pass

    def __str__(self):
        return "Universe Fuzzy Set"

    def __repr__(self):
        return str(self)

    def isInSupport(self, xi):
        return True

    def isFirstOfPartition(self):
        return True

    def isLastOfPartition(self):
        return True

    def membershipDegree(self, xi):
        return 1.0

    @staticmethod
    def createFuzzySet():
        return UniverseFuzzySet()